app/models/host/managed.rb

```diff
@@ -196,6 +196,7 @@ class Jail < ::Safemode::Jail
     validate :short_name_periods
     before_validation :set_compute_attributes, :on => :create, :if => Proc.new { compute_attributes_empty? }
     validate :check_if_provision_method_changed, :on => :update, :if => Proc.new { |host| host.managed }
+    validates :uuid, uniqueness: { :scope => :compute_resource_id }
```

Comment:
I'm undecided here. Both make sense. We actually do have several compute resources backed by the same VMWare infrastructure.